packages/staking/src/governance/api.ts

```diff
@@ -0,0 +1,102 @@
+import {GOVERNANCE_ENDPOINTS} from './config'
+import {DRepId} from './types'
+import {Fetcher, fetcher} from '@yoroi/common'
+
+export type GovernanceApi = {
+  getDRepById: (drepId: DRepId) => Promise<{txId: string; epoch: number} | null>
+  getStakingKeyState: (
+    stakeKeyHash: string,
+  ) => Promise<GetStakingKeyStateResponse>
+}
+
+export const governanceApiMaker = ({
+  networkId,
+  client = fetcher,
+}: {
+  networkId: number
+  client?: Config['client']
+}) => {
+  return new Api({networkId, client})
+}
+
+class Api implements GovernanceApi {
+  constructor(private config: Config) {}
+
+  async getDRepById(drepId: DRepId) {
+    const {networkId, client} = this.config
+    const backend = getApiConfig(networkId)
+
+    try {
+      const url = backend.getDRepById.replace('{{DREP_ID}}', drepId)
+      const response = await client<GetDRepByIdResponse>({url})
+      const txId = response?.registration?.tx
+      const epoch = response?.registration?.epoch
+      return txId && epoch ? {txId, epoch} : null
+    } catch (error) {
+      if (error instanceof Error && error.message.includes('404')) {
+        return null
+      }
+
+      throw error
+    }
+  }
+
+  async getStakingKeyState(stakeKeyHash: string) {
```

Comment:
since the config is a dependency of getStaking.. I'd curry it and return the function, while doing that for all methods on the factory creation. ie. https://github.com/Emurgo/yoroi/pull/2866/files#diff-8cb0399d9caf42c3a6d238998088ccc7af8300f3c6f12a40f3f7bf58ac825447R10 it also makes simpler to test.